Anime-App/Feature/Anime/View/ViewControllers/AnimeViewController.swift

```diff
@@ -117,59 +124,93 @@ extension AnimeViewController {
         createSearchBar()
         view.addSubview(tableView)
         self.tableView.addSubview(self.refreshControl)
+        
+        customLoadingView.center = self.view.center
+        self.view.addSubview(customLoadingView)
+        customLoadingView.bringSubviewToFront(self.view)
+        
         spinner.center = self.view.center
         self.view.addSubview(spinner)
         spinner.bringSubviewToFront(self.view)
-
+        
         NSLayoutConstraint.activate([
+            
+            customLoadingView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            customLoadingView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            
             spinner.centerXAnchor.constraint(equalTo: view.centerXAnchor),
-            spinner.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            spinner.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: 50),
+            
             tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
             tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 16),
             tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -16),
             tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
         ])
     }
     
-    func fetchAnime() {
-        self.isFetching = true
-        self.spinner.startAnimating()
+    
+    
+    
+    func resetAnime() {
+        animeViewModel.resetAnime { res in
+            switch res {
+            case .success:
+                self.tableView.reloadData()
+            case .failure:
+                print("Failed to reset")
+            }
+        }
+    }
+    
+    func startPagination() {
+        if isPaginating && !isFetching {
+             self.spinner.startAnimating()
+        }
+        isPaginating = true
         animeViewModel.fetchAnime(page: page)  { [weak self] res in
             switch res {
             case .success:
                 self?.tableView.reloadData()
-             case .failure:
+                self?.page+=1
+            case .failure:
                 print("no more data")
-             }
+            }
+            self?.isPaginating = false
             self?.spinner.stopAnimating()
-            self?.isFetching = false
         }
     }
     
-    func resetAnime() {
-        animeViewModel.resetAnime { res in
+    
+    func fetchAnime() {
+        if !isFetching && !isPaginating {
```

Comment:
This if statement is not needed at all, if you're fetching an anime then you want to just show the spinner and then stop it when you get a response back